monitor.go

```diff
@@ -37,18 +44,33 @@ func (m *defaultMonitor) Run(c *exec.Cmd) error {
 	return c.Run()
 }
 
-func (m *defaultMonitor) Start(c *exec.Cmd) error {
-	return c.Start()
-}
-
-func (m *defaultMonitor) Wait(c *exec.Cmd) (int, error) {
-	if err := c.Wait(); err != nil {
-		if exitErr, ok := err.(*exec.ExitError); ok {
-			if status, ok := exitErr.Sys().(syscall.WaitStatus); ok {
-				return status.ExitStatus(), nil
+func (m *defaultMonitor) Start(c *exec.Cmd) (chan Exit, error) {
+	ec := make(chan Exit, 1)
```

Comment:
nit: this can be moved just before the `go func` to avoid an extra `close(ec)` 2 lines below